### PR TITLE
Auctions

### PR DESCRIPTION
**Pull request by RobertMrowiec**
# Tasks:
- redirect from / to /auctions. This page has 3 auctions and login form below them

# Testing scenario:
go to / page

# Preview
image

### CODE REVIEW

File: app/templates/login.hbs [hunk 1]
```diff
@@ -1,2 +1,3 @@
+<Auctions class='auctions-container'></Auctions>
```
Comment (Rxbsxn) [inline app/templates/login.hbs]: Why do you render his component here?

File: app/templates/auctions.hbs [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<Auctions class='auctions-container'></Auctions>
```
[the remark this answers is not in the document]
Comment (RobertMrowiec) [reply app/templates/auctions.hbs]: Yea I just made it actually, but not pushed yet :P

File: app/templates/components/navbar.hbs [hunk 3]
```diff
@@ -10,16 +10,19 @@
...
 			<li class="nav-item">
-				<a class='nav-link'> Home </a>
+				{{#link-to 'login' class='nav-link'}} 
+					Home 
+				{{/link-to}}
 			</li>
 			<li class="nav-item">
-				<a class='nav-link'> Auctions </a>
+				{{#link-to 'auctions' class='nav-link'}} 
+					Auctions
```
Comment (Rxbsxn) [inline app/templates/components/navbar.hbs]: It's good, but for practice, you can read about LinkComponent in Ember and create a generic component with this specific class

File: app/templates/login.hbs (hunk 1 above)
Comment (RobertMrowiec) [reply app/templates/login.hbs]: Due to project flow: 

Website guest has limited view to 3 cards only, then down the below he should see buttons to register/login

**Comment by Rxbsxn** [comment]
You can create an acceptance test for the auctions route, setup ember-cli-mirage and start work with tests.

File: app/templates/login.hbs (hunk 1 above)
Comment (Rxbsxn) [reply app/templates/login.hbs]: @RobertMrowiec My bad I didn't see the screenshot, it's ok

File: app/components/login.js [hunk 4]
```diff
@@ -1,15 +1,28 @@
 import Component from '@ember/component';
+import { inject as service } from '@ember/service'
 
 export default Component.extend({
 	classNames: ['loginPage'],
 	email: '',
 	password: '',
+	isLogged: null,
+	auth: service(),
```
Comment (Rxbsxn) [inline app/components/login.js]: There is practice, how to structure components:
1) services
2) Elements of Component (tagName, classNames etc)
3) Initial properties
4) Short computed properties
5) Few lines computed properties
6) Tasks
7) Lifecycle methods
8) actions
9) private methods

File: tests/acceptance/login-test.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  test('visiting / should redirect to /login', async function(assert) {
+    await visit('/');
+
+    assert.equal(currentURL(), '/login');
+  });
+
+  test('click login button should redirect to /auction-list', async function(assert) {
+    await visit('/');
+    await click('[data-test-login-button]')
```
Comment (Rxbsxn) [inline tests/acceptance/login-test.js]: 👍

File: app/components/login.js (hunk 4 above)
Comment (RobertMrowiec) [reply app/components/login.js]: Soooo good to know that! Thank You!

Comment (RobertMrowiec) [reply app/components/login.js]: Should be there a new line between each type ? like:

```
asd: service(),
asd2: service(),

tagName: 'span',
classNames: 'defaultclass',

key: null,
anotherKey: null
```

or I should only use it only before actions?

Comment (Rxbsxn) [reply app/components/login.js]: It's good to have one empty between each type :+1:

File: app/models/car.js [hunk 6]
```diff
@@ -5,5 +5,6 @@ export default DS.Model.extend({
 	brand: DS.attr('string'),
 	horsePower: DS.attr('string'),
 	maxTorque: DS.attr('string'),
-	model: DS.attr('string')
+	model: DS.attr('string'),
+	// auction: DS.belongsTo('auction'),
```
Comment (Rxbsxn) [inline app/models/car.js]: :scissors:

File: mirage/config.js [hunk 7]
```diff
@@ -1,26 +1,31 @@
...
 
-  /*
-    Config (with defaults).
+	/*
+		Config (with defaults).
 
-    Note: these only affect routes defined *after* them!
-  */
+		Note: these only affect routes defined *after* them!
+	*/
```
Comment (Rxbsxn) [inline mirage/config.js]: What about these comments?

File: app/services/auth.js [hunk 8]
```diff
@@ -14,9 +14,10 @@ export default Service.extend({
 			this.set('correctCredentials', true)
 			return this.get('router').transitionTo('/auction-list')
 		})
-		.catch(reason => {
-			this.set('toastDescription', reason.text.replace(/"/g, ''))
-			return this.set('correctCredentials', false)
+		.catch(() => {
+			this.set('toastDescription', 'Wrong credentials')
+			this.set('correctCredentials', false)
+			return setTimeout(() => this.set('correctCredentials', null), 2000)
```
Comment (Rxbsxn) [inline app/services/auth.js]: https://api.emberjs.com/ember/3.9/functions/@ember%2Frunloop/later

File: app/styles/app.sass [hunk 9]
```diff
@@ -4,3 +4,4 @@
 @import 'components/auction-list.sass'
 @import 'components/login.sass'
 @import 'components/navbar.sass'
+@import 'components/toast.sass'
```
Comment (Rxbsxn) [inline app/styles/app.sass]: I think you don't need to pass a file extension here, you could verify it?

**Comment by Rxbsxn** [comment]
@RobertMrowiec Take a look at my latest comments. Then after applying potential feedback, we should move on and merge the pull-request into `loginPage` because there are many many changes here which make the pull-request hard to read and review.

File: app/services/auth.js [hunk 10]
```diff
@@ -12,12 +13,13 @@ export default Service.extend({
 		return this.get('session').authenticate('authenticator:jwt', credentials)
 		.then(() => {
 			this.set('correctCredentials', true)
-			return this.get('router').transitionTo('/auction-list')
+			this.get('router').transitionTo('/auction-list')
+			return later(() => this.set('correctCredentials', null), 2000)
```
Comment (Rxbsxn) [inline app/services/auth.js]: I think we can extract it to an extra method, due to using it in `then` and `catch`

**Pull request merged**